sanity: allow `Volume` `capacity_bytes` to be zero

NicolasT opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
3. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
From the CSI spec:

> The capacity of the volume in bytes. This field is OPTIONAL. If not
> set (value of 0), it indicates that the capacity of the volume is
> unknown (e.g., NFS share).
> The value of this field MUST NOT be negative.

As such, the tests that validate the result of a `CreateVolume`
operation shall accept zero as the `capacity_bytes` of the returned
`Volume` object, when validating this value against the requested
capacity.

See: https://github.com/container-storage-interface/spec/blob/e129a75169c13bade7c7336afe21f259740ba679/spec.md#createvolume

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Related to #94 but not fixing it completely (cfr. #263).

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
The tests that validate the result of a `CreateVolume` operation now accept zero as the `capacity_bytes` of the returned `Volume` object, when validating this value against the requested
capacity.
```

NicolasT commented:
/assign @jsafrane

msau42 commented:
/ok-to-test

msau42 commented:
/lgtm
/approve

Thanks for the fix! One last thing, can you add a release note for this fix? You can "/hold cancel" afterwards.
/hold

NicolasT commented:
Added a relnotes entry. Also, I removed the `Fixes` marker for #94 since this PR doesn't fix the whole issue (#263 plays a role as well).

/hold cancel
